janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java

```diff
@@ -696,13 +699,47 @@ public ModificationSummary prepareCommit(final Collection<InternalRelation> adde
         ListMultimap<InternalVertex, InternalRelation> mutatedProperties = ArrayListMultimap.create();
         List<IndexUpdate> indexUpdates = new ArrayList<>();
 
+        long startTimeStamp = System.currentTimeMillis();
+        int totalMutations = addedRelations.size() + deletedRelations.size();
+        boolean isBigDataSet = totalMutations >= backend.getBufferSize();
```

Comment:
I suggest changing it to:
```
boolean isBigDataSetLoggingEnabled = logForPrepareCommit.isDebugEnabled() && totalMutations >= backend.getBufferSize();
```
and replace usage of `isBigDataSet` to `isBigDataSetLoggingEnabled`.
That way the below string concatenation will be performed only when needed instead of each big transaction.

Reply:
Done